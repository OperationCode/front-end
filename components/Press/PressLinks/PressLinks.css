.logos a {
  color: #249cbc;
  text-decoration: none;
}

.logos .flexContainer {
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  padding: 15px 0 0 0;
}
