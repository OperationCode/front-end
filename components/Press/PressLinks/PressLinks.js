import React from 'react';
import { number } from 'prop-types';
import styles from './PressLinks.module.css';
import * as Articles from './Articles';
import ArticleGroup from './ArticleGroup/ArticleGroup';

PressLinks.propTypes = {
  numberOfInitiallyVisibleLinks: number,
};

PressLinks.defaultProps = {
  numberOfInitiallyVisibleLinks: 5,
};

function PressLinks({ numberOfInitiallyVisibleLinks }) {
  return (
    <div className={styles.logos}>
      <div className={styles.flexContainer}>
        {Object.keys(Articles).map(region => (
          <ArticleGroup
            key={`ArticleGroup_${region}`}
            region={region}
            articles={Articles[region]}
            numberOfInitiallyVisibleLinks={numberOfInitiallyVisibleLinks}
          />
        ))}
      </div>
    </div>
  );
}

export default PressLinks;
