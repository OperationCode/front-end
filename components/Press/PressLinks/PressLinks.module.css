.logos a {
  color: var(--primary);
  text-decoration: none;
}

.logos .flexContainer {
  width: 70ch;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 15px 0 0 0;
}

.logos .tabsList {
  display: flex;
  flex-shrink: 0;
}

.logos .tabsTrigger {
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: center;
  padding: 10px 20px 0;
  border-bottom: 2px solid rgba(var(--rgbValuesPrimary), 0.2);
}

.logos .tabsTrigger:hover {
  border-bottom: 2px solid rgba(var(--rgbValuesPrimary), 0.6);
  cursor: pointer;
}

.logos .tabsTrigger:focus-visible {
  outline: 1px solid var(--primary);
}

.logos .tabsTrigger[data-state="active"] {
  border-bottom: 2px solid var(--primary);
}

.logos .tabsContent {
  flex-grow: 1;
  padding: 20px;
}

.logos li {
  padding-bottom: 5px;
  text-align: center;
  list-style: none;
  font-size: 16px;
}

.logos li svg {
  top: 0;
  left: 5px;
  width: 0.6em;
}

.logos li::marker {
  display: none;
}

@media screen and (--medium-viewport) {
  .logos .flexContainer {
    width: 60ch;
  }
}

@media screen and (--small-viewport) {
  .logos {
    width: 80%;
  }

  .logos .flexContainer {
    width: 100%;
  }

  .logos .tabsList {
    flex-wrap: wrap;
  }

  .logos .tabsTrigger {
    min-width: 80%;
  }
}
