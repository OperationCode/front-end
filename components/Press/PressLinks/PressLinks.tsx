import * as Tabs from '@radix-ui/react-tabs';
import OutboundLink from 'components/OutboundLink/OutboundLink';
import { objectKeys } from 'utils/types';
import { cx } from 'common/utils/cva';
import * as Articles from './Articles';

function PressLinks() {
  return (
    <div className="w-full sm:w-[70ch] md:w-[60ch]">
      <Tabs.Root defaultValue={Object.keys(Articles)[0]}>
        <Tabs.List className="flex pt-5">
          {Object.keys(Articles).map(region => (
            <Tabs.Trigger
              key={`TabsTrigger_${region}`}
              value={region}
              className={cx(
                'flex-1 cursor-pointer',
                'border-b-2 border-primary/20 hover:border-primary/60',
                'data-[state=active]:border-primary',
              )}
            >
              <h3>{region}</h3>
            </Tabs.Trigger>
          ))}
        </Tabs.List>
        {objectKeys(Articles).map(region => (
          <Tabs.Content
            key={`TabsContent_${region}`}
            value={region}
            className="p-5 space-y-2.5 text-balance list-none text-center"
          >
            {Articles[region].map((link: { url: string; title: string }) => (
              <li key={`GroupLink_${link.url}`}>
                <OutboundLink
                  href={link.url}
                  analyticsEventLabel="Press Article"
                  className="text-sm text-primary no-underline transition-colors duration-200 ease-linear hover:text-white focus-visible:text-white"
                >
                  {link.title}
                </OutboundLink>
              </li>
            ))}
          </Tabs.Content>
        ))}
      </Tabs.Root>
    </div>
  );
}

export default PressLinks;
