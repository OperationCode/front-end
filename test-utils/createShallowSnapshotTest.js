import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers, components utilizing something that requires large
 * mocking, or very large components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
