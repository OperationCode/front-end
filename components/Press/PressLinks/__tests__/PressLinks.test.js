import React from 'react';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import PressLinks from '../PressLinks';

describe('PressLinks', () => {
  test('it should render properly no props', () => createShallowSnapshotTest(<PressLinks />));
});
