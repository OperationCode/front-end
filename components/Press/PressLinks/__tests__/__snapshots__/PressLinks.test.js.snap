// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressLinks should render with no props passed 1`] = `
<div
  className="logos"
>
  <div
    className="flexContainer"
  >
    <ArticleGroup
      articles={
        Array [
          Object {
            "title": "New program aims to help veterans land jobs in tech industry",
            "url": "https://wtkr.com/2018/05/16/new-program-aims-to-help-veterans-land-jobs-in-tech-industry/",
          },
          Object {
            "title": "Veterans in Residence",
            "url": "http://www.wdtv.com/content/news/Veterans-in-Residence-program-helping-veterans-grow-businesses-482707221.html",
          },
          Object {
            "title": "After Flying Solo, Veterans Find Others Who Have Their Backs",
            "url": "https://www.wework.com/blog/posts/after-flying-solo-veterans-find-others-who-have-their-back",
          },
        ]
      }
      numberOfInitiallyVisibleLinks={5}
      region="NYC"
    />
    <ArticleGroup
      articles={
        Array [
          Object {
            "title": "How Tech Pros Can Volunteer in Fun Ways",
            "url": "https://insights.dice.com/2017/01/11/tech-pros-volunteer-fun-ways/",
          },
          Object {
            "title": "MassChallenge Finalists: Finnest, Operation Code, Sunrise Health",
            "url": "https://www.necn.com/news/business/MassChallenge-Finnest_-Operation-Code_-Sunrise-Health_NECN-441063423.html",
          },
          Object {
            "title": "TechBreakfast at Microsoft NERD",
            "url": "http://www.wickedbandwidth.com/02072017-techbreakfast-microsoft-nerd/",
          },
          Object {
            "title": "Operation Code - #BNT80 Boston New Technology Startup Demo",
            "url": "https://www.youtube.com/watch?v=YGaG_HL3oEQ",
          },
          Object {
            "title": "MassChallenge Finalist | Operation Code | Conrad Hollomon",
            "url": "https://www.youtube.com/watch?v=GcvVsVsH-Gg",
          },
          Object {
            "title": "Operation Code – Creating a Coding Community with Veterans and Citizens",
            "url": "https://venturefizz.com/stories/boston/operation-code-creating-coding-community-veterans-and-citizens",
          },
          Object {
            "title": "Innovation Spotlight: Operation Code",
            "url": "https://www.deepcoredata.com/innovation-spotlight-veterans-mission-operation-code/",
          },
          Object {
            "title": "tech4vets Masschallenge Veterans",
            "url": "https://blogs.microsoft.com/newengland/2017/11/10/tech4vets-masschallenge-veterans/",
          },
        ]
      }
      numberOfInitiallyVisibleLinks={5}
      region="Boston"
    />
    <ArticleGroup
      articles={
        Array [
          Object {
            "title": "TechHire Educator Spotlight: Operation Code",
            "url": "https://blog.opportunityatwork.org/techhire-educator-spotlight-operation-code-debd0a796f9d",
          },
          Object {
            "title": "Why Veterans Will Make Excellent Programmers",
            "url": "https://www.switchup.org/blog/why-veterans-will-make-excellent-programmers",
          },
          Object {
            "title": "THOUSANDS OF VETERANS WANT TO LEARN TO CODE — BUT CAN’T",
            "url": "https://www.wired.com/story/thousands-of-veterans-want-to-learn-to-code-but-cant/",
          },
          Object {
            "title": "Hacking Entrepreneurship — An Interview with David Molina of Operation Code",
            "url": "https://medium.com/the-hum/hacking-entrepreneurship-an-interview-with-david-molina-of-operation-code-12a7e199e4e0",
          },
          Object {
            "title": "Operation Code: Connecting Veterans with Code Skills",
            "url": "http://blog.teamtreehouse.com/operation-code-connecting-veterans-with-code",
          },
          Object {
            "title": "Operation Code: connecting tech and veterans",
            "url": "https://blog.github.com/2016-11-11-operation-code-connecting-tech-and-veterans/",
          },
          Object {
            "title": "When the call of duty is technology, veterans rally to support each other through Operation Code",
            "url": "https://www.geekwire.com/2016/call-duty-technology-veterans-rally-support-operation-code/",
          },
          Object {
            "title": "Get Coding Now with Operation Code Army Veteran and Founder David Molina",
            "url": "https://www.youtube.com/watch?v=qoy7scC2SHk",
          },
          Object {
            "title": "The New Developer - Operation Code - GitHub Universe 2016",
            "url": "https://www.youtube.com/watch?v=xN7yMoe38xc",
          },
          Object {
            "title": "What happens when military veterans learn to code - CodeConf 2016",
            "url": "https://www.youtube.com/watch?v=-wSwlLeKFdE",
          },
          Object {
            "title": "How Operation Code helps veterans learn programming skills",
            "url": "https://opensource.com/article/17/4/operation-code-mentors-veterans",
          },
          Object {
            "title": "Helping Veterans Learn to Code with David Molina",
            "url": "https://softwareengineeringdaily.com/2016/03/20/helping-veterans-learn-code-david-molina/",
          },
          Object {
            "title": "Podcast: a developing story as told by developers",
            "url": "https://developingstory.netlify.com/tds-66-david-molina-from-operation-code",
          },
          Object {
            "title": "Instagram: David Molina",
            "url": "https://www.instagram.com/p/21p5bFxUjd/?taken-by=davidcmolina",
          },
          Object {
            "title": "From Aviation Electrician to Back End Engineering: Bret Funk’s Operation Code story",
            "url": "http://blog.teamtreehouse.com/from-aviation-electrician-to-back-end-engineering-bret-funk-operation-code-story",
          },
          Object {
            "title": "Navy Veteran to Software Developer: Geno Guerrero’s Operation Code Story",
            "url": "http://blog.teamtreehouse.com/navy-veteran-to-software-developer-geno-guerreros-operation-code-story",
          },
          Object {
            "title": "From Marine Corps Veteran to Front End Developer: Billy Le’s Operation Code Story",
            "url": "http://blog.teamtreehouse.com/marine-corps-veteran-front-end-developer-billy-le",
          },
          Object {
            "title": "Few Options For Veterans Looking To Enter Tech",
            "url": "https://techcrunch.com/2015/11/11/few-options-for-veterans-looking-to-enter-tech/",
          },
          Object {
            "title": "Coding Boot Camps Go After Veterans To Take Silicon Valley's Vacant Tech Jobs",
            "url": "https://www.ibtimes.com/coding-boot-camps-go-after-veterans-take-silicon-valleys-vacant-tech-jobs-2174421",
          },
          Object {
            "title": "20 Diversity and Inclusion Leaders to Follow in 2018",
            "url": "https://www.makeschool.com/blog/20-diversity-and-inclusion-leaders-to-follow-in-2018",
          },
          Object {
            "title": "Tech innovation meets military service: GeekWire’s Memorial Day remembrance and update",
            "url": "http://bitshare.cm/news/tech-innovation-meets-military-service-geekwires-memorial-day-remembrance-and-update/",
          },
          Object {
            "title": "MAP HOSTS CONGRESSIONAL BRIEFING ON VETERANS' READINESS IN TECH CAREERS",
            "url": "https://www.millennialaction.org/press-archives/vetstechpr",
          },
          Object {
            "title": "Analytics Pros, Inc. Hosts Training for Veterans and Veterans' Spouses",
            "url": "https://www.advfn.com/news_Analytics-Pros-Inc-Hosts-Training-for-Veterans-a_72657464.html",
          },
          Object {
            "title": "Episode 36 – How to pick a programming language to learn for new developers – Part 2",
            "url": "http://mentoringdevelopers.com/episode-36-how-to-pick-a-programming-language-to-learn-for-new-developers-part-2/",
          },
          Object {
            "title": "How to “upskill” Veterans’ training for technology jobs",
            "url": "https://www.blogs.va.gov/VAntage/33028/upskill-veterans-training-technology-jobs/",
          },
          Object {
            "title": "Instagram: Operation Code made the front cover of Oregon Veterans News Magazine",
            "url": "https://www.instagram.com/p/BbZT3Q-BBDC/",
          },
          Object {
            "title": "Jameel: From Marines to SoftwareEngineer",
            "url": "https://blog.sabio.la/jameel-from-jarhead-to-softwareengineer-9130702da5c1",
          },
          Object {
            "title": "Seattle coding-school tuition to be covered by GI Bill",
            "url": "https://www.seattletimes.com/business/technology/seattles-code-fellows-wins-approval-under-gi-bill/",
          },
          Object {
            "title": "Veteran: GI Bill should cover code school",
            "url": "https://www.koin.com/news/veteran-gi-bill-should-cover-code-school_20180208085618706/960235415",
          },
          Object {
            "title": "Patriot Boot Camp And Operation Code Join Forces To Help Military Veterans Become Technology Entrepreneurs",
            "url": "http://patriotbootcamp.org/blog/operation-code",
          },
          Object {
            "title": "Operation Code wants veterans to work in tech",
            "url": "http://www.maswired.com/operation-code-wants-veterans-to-work-in-tech/",
          },
          Object {
            "title": "Coding Bootcamps Accepts GI Bill",
            "url": "http://diversemilitary.net/2017/06/27/coding-bootcamps-accepts-gi-bill/",
          },
          Object {
            "title": "Operation Code Looks to Help Veterans Land IT Careers",
            "url": "https://news.clearancejobs.com/2017/07/17/operation-code-looks-help-veterans-land-careers/",
          },
        ]
      }
      numberOfInitiallyVisibleLinks={5}
      region="General"
    />
  </div>
</div>
`;
